### PR TITLE
update unit tests to use testify package

### PR DESCRIPTION
**Pull request by jy19**
### Summary

This PR updates some unit tests to use the `github.com/stretchr/testify` package as requested in this issue: https://github.com/aws/amazon-ecs-init/issues/98

### Testing

`make gotest` runs successfully 

New tests cover the changes: 
no additional tests added 

### Description for the changelog

Enhancement - update some ecs-init unit tests to use the testify package 

### Licensing

This contribution is under the terms of the Apache 2.0 License:

### CODE REVIEW

File: ecs-init/cache/cache_test.go [hunk 1]
```diff
@@ -213,9 +202,7 @@ func TestDownloadAgentDownloadMD5Failure(t *testing.T) {
 	mockMetadata := NewMockinstanceMetadata(mockCtrl)
 
 	tempMD5File, err := ioutil.TempFile("", "md5-test")
-	if err != nil {
-		t.Fail()
-	}
+	assert.True(t, err == nil, "Expect to successfully create a temporary file")
```
Comment (cyastella) [inline ecs-init/cache/cache_test.go]: Maybe assert.NoError()? Same below.

File: ecs-init/backoff/backoff_test.go [hunk 2]
```diff
@@ -16,48 +16,33 @@ package backoff
...
 )
 
 func TestAddJitter(t *testing.T) {
 	for i := 0; i < 10; i++ {
 		duration := addJitter(10*time.Second, 3*time.Second)
-		if duration < 10*time.Second || duration > 13*time.Second {
-			t.Error("Excessive amount of jitter", duration)
-		}
+
+		assert.False(t, duration < 10*time.Second, "Expect jitter > 10*time.Second")
```
Comment (cyastella) [inline ecs-init/backoff/backoff_test.go]: nit: Expect jitter >=, similar below.

File: ecs-init/cache/cache_test.go (hunk 1 above)
Comment (fierlion) [reply ecs-init/cache/cache_test.go]: I'm not clear why we open the `tempMD5File` here in the first place.  This test shouldn't be testing ioutil.TempFile().  Maybe this isn't the PR for this question.

I see assert.NoError() being used as a conditional in the docs:
```
  actualObj, err := SomeFunction()
  if assert.NoError(t, err) {
	   assert.Equal(t, expectedObj, actualObj)
  }
```
and if we use it like this, I'd actually prefer the `assert.True()`
And we still need to defer closing this `tempMD5File` below.

File: ecs-init/backoff/backoff_test.go [hunk 3]
```diff
@@ -16,48 +16,33 @@ package backoff
...
-	}
-	// Expect next backoff duration to be max backoff
-	if duration := retryBackoff.Duration(); duration != 3*time.Second {
-		t.Error("Increase in duration is incorrect, exceeded maximum")
-	}
-	if retryBackoff.ShouldRetry() != false {
-		t.Errorf("Expected to not retry when count >= max retries")
-	}
+	assert.Equal(t, retryBackoff.Duration(), time.Second, "expect initial backoff to equal min backoff duration")
+	assert.Equal(t, retryBackoff.Duration(), 2*time.Second, "expect second back off to equal 2 * min backoff")
```
Comment (yumex93) [inline ecs-init/backoff/backoff_test.go]: nit: second backoff

File: ecs-init/backoff/backoff_test.go [hunk 4]
```diff
@@ -16,48 +16,33 @@ package backoff
...
+	assert.Equal(t, retryBackoff.Duration(), 3*time.Second, "expect 3rd back off to be max backoff")
```
Comment (yumex93) [inline ecs-init/backoff/backoff_test.go]: nit: same here. backoff

File: ecs-init/cache/cache_test.go (hunk 1 above)
Comment (yumex93) [reply ecs-init/cache/cache_test.go]: I did not see tempMD5File is used in this test case. You may want to just remove it.

File: ecs-init/cache/cache_test.go [hunk 5]
```diff
@@ -242,9 +229,7 @@ func TestDownloadAgentReadPublishedMd5Failure(t *testing.T) {
...
```
Comment (yumex93) [inline ecs-init/cache/cache_test.go]: assert.NoError would be a conciser choice.

File: ecs-init/docker/dependencies_test.go [hunk 6]
```diff
@@ -32,27 +34,19 @@ var netError = &url.Error{Err: &net.OpError{Op: "read", Net: "unix", Err: io.EOF
 var httpError = &docker.Error{Status: http.StatusInternalServerError, Message: "error"}
 
 func TestIsNetworkErrorReturnsTrue(t *testing.T) {
-	if !isNetworkError(netError) {
-		t.Errorf("Expected true when checking if network error")
-	}
+	assert.True(t, isNetworkError(netError), "Expect network error to be true if netError passed in")
```
Comment (yumex93) [inline ecs-init/docker/dependencies_test.go]: nit: The error message is a little bit confusing. Maybe "Expect isNetworkError to return true if network error passed in"?

File: ecs-init/docker/dependencies_test.go [hunk 7]
```diff
@@ -32,27 +34,19 @@ var netError = &url.Error{Err: &net.OpError{Op: "read", Net: "unix", Err: io.EOF
...
 }
 
 func TestIsNetworkErrorReturnsFalse(t *testing.T) {
-	if isNetworkError(fmt.Errorf("error")) {
-		t.Errorf("Expected false when checking if network error")
-	}
+	assert.False(t, isNetworkError(fmt.Errorf("error")), "Expect network error to be false non-network error passed in")
```
Comment (yumex93) [inline ecs-init/docker/dependencies_test.go]: Same here. "Expect isNetworkError to return false if non-network error passed in"?

File: ecs-init/docker/dependencies_test.go [hunk 8]
```diff
@@ -72,9 +66,7 @@ func TestNewDockerClientRetriesOnPingNetworkError(t *testing.T) {
 	)
 
 	_, err := newDockerClient(mockClientFactory, mockBackoff)
-	if err != nil {
-		t.Error("Error creating docker client")
-	}
+	assert.False(t, err != nil, "Expect no error for creating docker client with retry on network error")
```
Comment (yumex93) [inline ecs-init/docker/dependencies_test.go]: assert.NoError probably is better.

File: ecs-init/docker/dependencies_test.go [hunk 9]
```diff
@@ -149,9 +135,8 @@ func TestNewDockerClientGivesUpRetryingOnUnavailableSocket(t *testing.T) {
 	mockBackoff := NewMockBackoff(ctrl)
 
 	realDockerClient, dockerClientErr := godockerClientFactory{}.NewVersionedClient("unix:///a/bad/docker.sock", dockerClientAPIVersion)
-	if dockerClientErr != nil {
-		t.Fatal("error setting up intentionally bad client (nonexistent socket path)")
-	}
+
+	require.False(t, dockerClientErr != nil, "there should be no errors trying to set up a docker client (intentionally bad with nonexistent socket path")
```
Comment (yumex93) [inline ecs-init/docker/dependencies_test.go]: require.NoError?

File: ecs-init/cache/cache_test.go (hunk 1 above)
Comment (jy19) [reply ecs-init/cache/cache_test.go]: i removed this since it's not used

**Comment by jy19** [comment]
addressed nits and changed assert.True or assert.False regarding errors to assert.Error and assert.NoError

**Pull request merged**